### ISSUE TITLE
Follow overlay is missing input labels

### ISSUE THREAD

**Issue by MatzeKitt** [Automattic/wordpress-activitypub#555]
### Quick summary

If I use the follow me block and open the overlay, I noticed that there are no labels available. As regular user, I’m fine with that, but at least for accessibility there should be invisible labels for screen readers.

### Steps to reproduce

1. Create a page with the "Follow me" block.
2. Open the page
3. Open the overlay
4. Look into the source code

### What you expected to happen

A label for each input field is available, especially for screen readers.

### What actually happened

The label for the input fields are missing.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by mattwiebe** [comment]
Thanks for opening the issue @MatzeKitt 👍🏻
